crux_http/tests/with_tester.rs

```diff
@@ -1,87 +1,77 @@
 mod shared {
-    use crux_core::{capability::CapabilityContext, App, WithContext};
+    use crux_core::capability::CapabilityContext;
     use crux_http::{Http, HttpRequest};
+    use crux_macros::Effect;
     use serde::{Deserialize, Serialize};
     use url::Url;
 
     #[derive(Default)]
-    pub(crate) struct MyApp;
+    pub(crate) struct App;
 
     #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
-    pub enum MyEvent {
+    pub enum Event {
         Get,
         Set(String),
     }
 
     #[derive(Default, Serialize, Deserialize)]
-    pub struct MyModel {
+    pub struct Model {
         pub body: String,
     }
 
     #[derive(Serialize, Deserialize, Default)]
-    pub struct MyViewModel {
+    pub struct ViewModel {
         pub result: String,
     }
 
-    impl App for MyApp {
-        type Event = MyEvent;
-        type Model = MyModel;
-        type ViewModel = MyViewModel;
+    impl crux_core::App for App {
+        type Event = Event;
+        type Model = Model;
+        type ViewModel = ViewModel;
 
-        type Capabilities = MyCapabilities;
+        type Capabilities = Capabilities;
 
-        fn update(&self, event: MyEvent, model: &mut MyModel, caps: &MyCapabilities) {
+        fn update(&self, event: Event, model: &mut Model, caps: &Capabilities) {
             match event {
-                MyEvent::Get => {
+                Event::Get => {
                     caps.http
-                        .get_json(Url::parse("http://example.com").unwrap(), MyEvent::Set);
+                        .get_json(Url::parse("http://example.com").unwrap(), Event::Set);
                 }
-                MyEvent::Set(body) => {
+                Event::Set(body) => {
                     model.body = body;
                 }
             }
         }
 
         fn view(&self, model: &Self::Model) -> Self::ViewModel {
-            MyViewModel {
+            ViewModel {
                 result: format!("Body: {}", model.body),
             }
         }
     }
 
-    #[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
-    pub enum MyEffect {
-        Http(HttpRequest),
-    }
-
-    pub(crate) struct MyCapabilities {
-        pub http: Http<MyEvent>,
-    }
-
-    impl WithContext<MyApp, MyEffect> for MyCapabilities {
-        fn new_with_context(context: CapabilityContext<MyEffect, MyEvent>) -> MyCapabilities {
-            MyCapabilities {
-                http: Http::new(context.with_effect(MyEffect::Http)),
-            }
-        }
+    #[derive(Effect)]
+    pub(crate) struct Capabilities {
+        #[effect(operation = "HttpRequest")]
```

Comment:
Having to specify the operation here is a bit annoying.  Wonder if we could add an associated type to the `Capability` trait and just get it from there?

Suppose we'd need some way to differentiate between capabilities with data and those without, but reckon it might be do-able.

Suppose it could also make having a capability support >1 operation painful, although maybe we cross that bridge when we come to it?

Reply:
Yeah, still not sure how we'd get it though, if it's not in the input token stream?

Reply:
ahh just read the comment below